### PR TITLE
#163114860 Create Report Model 

### PR DESCRIPTION
**Pull request by Easybuoy**
#### What does this PR do?
Create a Report Model and Migration

#### Description of Task to be completed?
- Create `report` migration flie in `migration` directory and add the migrations.
- Create `report` model file in `model` directory and add the model.

#### How should this be manually tested?
- run `npm run db:migrate`
- check `database` to see `report` table amongst table created.

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?
#163114860

#### Questions:

### CODE REVIEW

File: models/report.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+    articleId: {
+      type: DataTypes.INTEGER,
+      allowNull: false
+
+    },
+  });
+    // Report.associate = models => Report.hasOne(models.Profile, {
```
Comment (oreoluwade) [inline models/report.js]: Why are these parts commented?

Comment (oreoluwade) [reply models/report.js]: I imagine there's an association between The `user and the report` as well as between the `article and the report`, at least based on the `userId` and `articleId` fields on the report model. Those relationships are not reflected here. Is there a reason why?

Comment (Easybuoy) [reply models/report.js]: > Why are these parts commented?

The comment was a mistake, I would remove them.

Comment (Easybuoy) [reply models/report.js]: > I imagine there's an association between The `user and the report` as well as between the `article and the report`, at least based on the `userId` and `articleId` fields on the report model. Those relationships are not reflected here. Is there a reason why?

The relationships are:
`USER -> HAS MANY -> REPORTS`
`ARTICLE -> HAS MANY -> REPORTS`
So i thought they would have to be defined in the `user` and `article` models and not in the `report` model. That was why I did not define the associations in the `report` model.

Comment (oreoluwade) [reply models/report.js]: Okay. You can add those relationships in the already existing related files then. 

Also, I'd like you to help change the structure of our migration files naming.

replace the numerical parts with `001`, `002` and so on till you get to the report migration file which will be `006`

Comment (Easybuoy) [reply models/report.js]: I've made requested changes and added the relationships.

Comment (oreoluwade) [reply models/report.js]: Okay.

The commented code is still in the report model however.

Comment (Easybuoy) [reply models/report.js]: Forgot again, removed and pushed.

**Review by oreoluwade — DISMISSED** [review]
LGTM

File: models/report.js (hunk 1 above)
Comment (Easybuoy) [reply models/report.js]: Travis failed, I am trying to find out why.

Comment (oreoluwade) [reply models/report.js]: It seems the `Reports` table is not being created when migrations are done.

**Comment by oreoluwade** [comment]
Meanwhile, please add these two scripts to package.json:
 `"db:migrate:testdb": "NODE_ENV=test sequelize db:migrate",`
 `"undo:migrations:testdb": "NODE_ENV=test sequelize db:migrate:undo:all"`

**Comment by Easybuoy** [comment]
> Meanwhile, please add these two scripts to package.json:
> `"db:migrate:testdb": "NODE_ENV=test sequelize db:migrate",`
> `"undo:migrations:testdb": "NODE_ENV=test sequelize db:migrate:undo:all"`

Okay

File: models/report.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+module.exports = (sequelize, DataTypes) => {
+  const Report = sequelize.define('Report', {
+    report: {
+      type: DataTypes.TEXT,
+      allowNull: false,
+    },
+    userId: {
+      type: DataTypes.STRING,
```
Comment (oreoluwade) [inline models/report.js]: This is where the problem is. The `userId` type should be `INTEGER`, not `STRING`

File: models/report.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+      allowNull: false,
+    },
+    articleId: {
+      type: DataTypes.INTEGER,
+      allowNull: false
+
```
Comment (oreoluwade) [inline models/report.js]: Remove this empty line.

File: models/report.js (hunk 1 above)
Comment (oreoluwade) [reply models/report.js]: > It seems the `Reports` table is not being created when migrations are done.

This was a test-related thing. Disregard.

Comment (Easybuoy) [reply models/report.js]: The `Reports` table is created when migrations are done.
I traced the travis error to the relationship i defined in the user model.
`User.hasMany(models.Report, {`
      `foreignKey: 'userId',`
      `onDelete: 'CASCADE'`
    });`

File: models/report.js (hunk 2 above)
Comment (Easybuoy) [reply models/report.js]: wow, thanks.

File: models/report.js (hunk 3 above)
Comment (Easybuoy) [reply models/report.js]: Done

File: models/report.js (hunk 2 above)
Comment (Easybuoy) [reply models/report.js]: I've pushed the correction

**Review by oreoluwade — APPROVED** [review]

**Pull request merged**